src/en/mangamo/src/eu/kanade/tachiyomi/extension/en/mangamo/Mangamo.kt

```diff
@@ -0,0 +1,435 @@
+            fields += SeriesDto::onlyTransactional.name
+        }
+
+        if (exclusivesOnlyPref.contains(MangamoConstants.EXCLUSIVES_ONLY_OPTION_IN_BROWSE)) {
+            fields += SeriesDto::onlyOnMangamo.name
+        }
+
+        orderBy = listOf(descending(SeriesDto::updatedAt.name))
+
+        // Filters can't be used with orderBy because firebase wants there to be indexes
+        // on various fields to support those queries and we can't create them.
+        // Therefore, all filtering has to be done on the client in the parse method.
+    }
+
+    override fun latestUpdatesParse(response: Response): MangasPage = parseMangaPage(response) {
+        if (it.enabled != true) {
+            return@parseMangaPage false
+        }
+        if (coinMangaPref.contains(MangamoConstants.HIDE_COIN_MANGA_OPTION_IN_BROWSE)) {
+            if (it.onlyTransactional == true) {
+                return@parseMangaPage false
+            }
+        }
+        if (exclusivesOnlyPref.contains(MangamoConstants.EXCLUSIVES_ONLY_OPTION_IN_BROWSE)) {
+            if (it.onlyOnMangamo != true) {
+                return@parseMangaPage false
+            }
+        }
+        true
+    }
+
+    // Search manga
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request = firestore.getCollection("Series") {
+        limit = MangamoConstants.BROWSE_PAGE_SIZE
+        offset = (page - 1) * MangamoConstants.BROWSE_PAGE_SIZE
+
+        val fields = seriesRequiredFields.toMutableList()
+        this.fields = fields
+
+        if (coinMangaPref.contains(MangamoConstants.HIDE_COIN_MANGA_OPTION_IN_BROWSE)) {
+            fields += SeriesDto::onlyTransactional.name
+        }
+
+        if (exclusivesOnlyPref.contains(MangamoConstants.EXCLUSIVES_ONLY_OPTION_IN_SEARCH)) {
+            fields += SeriesDto::onlyOnMangamo.name
+        }
+
+        // Adding additional filters makes Firestore complain about wanting an index
+        // so we filter on the client in parse, just like for Latest.
+
+        filter = and(
+            isEqual(SeriesDto::enabled.name, true),
+            isGreaterThanOrEqual(SeriesDto::name_lowercase.name, query.lowercase()),
+            isLessThanOrEqual(SeriesDto::name_lowercase.name, query.lowercase() + "\uf8ff"),
+        )
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage = parseMangaPage(response) {
+        if (coinMangaPref.contains(MangamoConstants.HIDE_COIN_MANGA_OPTION_IN_SEARCH)) {
+            if (it.onlyTransactional == true) {
+                return@parseMangaPage false
+            }
+        }
+        if (exclusivesOnlyPref.contains(MangamoConstants.EXCLUSIVES_ONLY_OPTION_IN_SEARCH)) {
+            if (it.onlyOnMangamo != true) {
+                return@parseMangaPage false
+            }
+        }
+        true
+    }
+
+    // Manga details
+
+    override fun getMangaUrl(manga: SManga): String {
+        return manga.url
```

Comment:
would be better to store relative urls in the database

Reply:
Converted to relative URLs